src/main/java/com/exasol/adapter/sql/UserInformation.java

```diff
@@ -0,0 +1,19 @@
+package com.exasol.adapter.sql;
+
+import com.exasol.adapter.jdbc.RemoteMetadataReaderException;
+
+import java.sql.*;
+
+public class UserInformation {
+    public int getRoleMask(final Connection connection) {
+        final String rlsUsersTableName = "rls_users";
+        final String query = "SELECT exa_role_mask FROM " + rlsUsersTableName + " WHERE exa_user_name = CURRENT_USER";
+        try (final ResultSet resultSet = connection.prepareStatement(query).executeQuery()) {
+            return resultSet.getInt("exa_role_mask");
```

Comment:
`next()` call missing.
Also please add a check that makes sure the result set has exactly one entry.
Otherwise set the mask to zero (safe default).
In case more then one entries exist additionally create log message on level `WARNING`.

Reply:
done